qa-tests-backend/src/test/groovy/ExternalIpFlowsTest.groovy

```diff
@@ -0,0 +1,197 @@
+
+    static final private int RETRY_COUNT = 5
+    static final private int RETRY_INTERVAL = 30
+
+    static final private String DEFAULT_QUERY = "Namespace:qa+Discovered External Source:true"
+
+    // Target deployments
+    @Shared
+    private List<Deployment> targetDeployments
+
+    // Source deployments
+    @Shared
+    private List<Deployment> sourceDeployments
+
+    def buildSourceDeployments() {
+        return [
+            new Deployment()
+                    .setName(EXTERNALDESTINATION)
+                    .setImage("quay.io/rhacs-eng/qa-multi-arch:nginx-1-15-4-alpine")
+                    .addLabel("app", EXTERNALDESTINATION)
+                    .setCommand(["/bin/sh", "-c",])
+                    .setArgs(["while sleep ${NetworkGraphUtil.NETWORK_FLOW_UPDATE_CADENCE_IN_SECONDS}; " +
+                                      "do wget -S -T 2 http://www.google.com; " +
+                                      "done" as String,]),
+            new Deployment()
+                    .setName("${EXTERNALDESTINATION}-2")
+                    .setImage("quay.io/rhacs-eng/qa-multi-arch:nginx-1-15-4-alpine")
+                    .addLabel("app", EXTERNALDESTINATION)
+                    .setCommand(["/bin/sh", "-c",])
+                    .setArgs(["while sleep ${NetworkGraphUtil.NETWORK_FLOW_UPDATE_CADENCE_IN_SECONDS}; " +
+                                      "do wget -S -T 2 http://www.httpbin.org; " +
+                                      "done" as String,]),
+        ]
+    }
+
+    @Shared
+    private List<Deployment> deployments
+
+    def createDeployments() {
+        deployments = buildSourceDeployments()
+        orchestrator.batchCreateDeployments(deployments)
+        for (Deployment d : deployments) {
+            assert Services.waitForDeployment(d)
+        }
+    }
+
+    def setupSpec() {
+        CollectorUtil.enableExternalIps(orchestrator)
+        createDeployments()
+    }
+
+    def destroyDeployments() {
+        for (Deployment deployment : deployments) {
+            orchestrator.deleteDeployment(deployment)
+        }
+        for (Deployment deployment : deployments) {
+            if (deployment.exposeAsService) {
+                orchestrator.waitForServiceDeletion(new Service(deployment.name, deployment.namespace))
+            }
+        }
+    }
+
+    def cleanupSpec() {
+        CollectorUtil.deleteRuntimeConfig(orchestrator)
+        destroyDeployments()
+    }
+
+    @Tag("NetworkFlowVisualization")
+    @IgnoreIf({ Env.REMOTE_CLUSTER_ARCH == "ppc64le" || Env.REMOTE_CLUSTER_ARCH == "s390x" })
+    def "Verify external network flow metadata"() {
+        when:
+
+        def metadataResponse
+        withRetry(RETRY_COUNT, RETRY_INTERVAL) {
+            metadataResponse = NetworkGraphService.getExternalNetworkFlowsMetadata(
+                DEFAULT_QUERY,
+            )
+            assert metadataResponse.getTotalEntities() == 2
```

Comment:
Perhaps first assert that metadataResponse is not null.

Reply:
Resolving this conversation since this has moved to a different PR :)